vm/qemu/qemu.go

```diff
@@ -534,7 +534,26 @@ func splitArgs(str, templateDir string, index int) (args []string) {
 		if arg == "" {
 			continue
 		}
-		arg = strings.ReplaceAll(arg, "{{INDEX}}", fmt.Sprint(index))
+		// 0-7 VFs/Device, if index >= 8, Device# += (index / 8)
+		if strings.Contains(arg, "{{INDEX}}") {
+			arg = strings.ReplaceAll(arg, "{{INDEX}}", fmt.Sprint(index%8))
+			if index > 7 && strings.Contains(arg, "vfio-pci,host=") {
+				// VFs are 0:7. Need to increment the Device if VF >= 8
+				args1 := strings.Split(arg, "vfio-pci,host=")
+				args2 := strings.Split(args1[1], ":")
+				args3 := strings.Split(args2[1], ".")
+				n, err := strconv.ParseInt(args3[0], 16, 32)
+				if err != nil {
+					fmt.Println(err)
+					return
+				}
+				i64 := int64(index)
+				n += (i64 / 8)
+				pat := fmt.Sprintf("%s:%s", args2[0], args3[0])
+				fix := fmt.Sprintf("%s:%02x", args2[0], n)
+				arg = strings.ReplaceAll(arg, pat, fix)
+			}
+		}
```

Comment:
This function looks too complex now. Lets test it. Tests will also better explain your intention.

Reply:
Worked with Dmitry awhile ago on the initial implementation, but it only handled 8 VFs. The above patch gets past the 8 VF limitation. The above has been successfully tested with 18 VMs with a unique VF assigned to each.

Reply:
dvyukov also added few comments. I still believe it worth testing:
1. To double check corner cases.
2. To simplify future modification preventing regressions.